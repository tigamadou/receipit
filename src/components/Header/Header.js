import React from 'react';
import PropTypes from 'prop-types';
import {
  Link,
} from 'react-router-dom';
import { FiArrowLeft } from 'react-icons/fi';
import Color from '../../styles/modules/Colors.module.scss';
import Font from '../../styles/modules/Font.module.scss';
import Background from '../../styles/modules/Background.module.scss';

const Header = ({ data }) => (

  <header className={`header ${Background.dark}`}>
    {data.back
          && (
            <Link to={data.back} className={`nav ${Color.white} ${Font.is_md}`}>
              <FiArrowLeft />
            </Link>
          )}
    <div className={`title ${Color.white} ${Font.is_sm}`}><h1>{data.title}</h1></div>
  </header>

);

Header.propTypes = {
  data: PropTypes.instanceOf(Object),
};
Header.defaultProps = {
  data: { back: false, title: 'Recipely', search: false },
};
export default Header;
