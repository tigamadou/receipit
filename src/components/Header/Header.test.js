import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Header from './Header';

describe('Header component tests', () => {
  const data = { title: 'Recipely', back: false };
  test('renders Header component ', () => {
    render(<Header data={data} />);
    // eslint-disable-next-line no-unused-expressions
    expect(screen.getByText('Recipely')).toBeInTheDocument;
  });

  test('should match with snapshot', () => {
    const tree = renderer
      .create(<Header data={data} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
