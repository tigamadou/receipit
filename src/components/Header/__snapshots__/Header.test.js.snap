// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component tests should match with snapshot 1`] = `
<header
  className="header dark"
>
  <div
    className="title white is_sm"
  >
    <h1>
      Recipely
    </h1>
  </div>
</header>
`;
